bin/ddg2logo

```diff
@@ -60,169 +55,53 @@ DESTABILIZING MUTATIONS
 This allows to obtain a 0-based positive-only plot in which the ratio between
 the energy values is the same as the original data."""
 
-def parse_letters_png(pngfile, letters="ACDEFGHIKLMNPQRSTVWXYypshzkmloef"):
-    if type(pngfile) is tuple:
-        pngfile = BytesIO(base64.b64decode("".join(pngfile)))
+def plot_logo(array,reslist,xlabels,splice,color_scheme,font,degeneracy):
 
-    png = imread(pngfile)
-    side_length = png.shape[0]
-    if png.shape[1] % png.shape[0] != 0:
-        log.error("Input PNG file doesn't have the right shape (horizontal size is expected to be a multplie of vertical size)")
-        raise TypeError
+    # Extract the array and labels
+    array_data = array     # 2D array
+    labels = xlabels       # List of labels
 
-    if png.shape[1] // png.shape[0] != len(letters):
-        log.error("Input PNG file does not contain the right number of letters")
-        raise TypeError
+    # Convert to DataFrame and assign labels as columns
+    df = pd.DataFrame(array_data, columns=reslist)
 
-    out = {}
-    for i,l in enumerate(letters):
-        out[l] = png[:,i*side_length:(i+1)*side_length,:]
+    # Add a column for the X letter for the degeneracy option
+    df['X'] = 0.00000
 
-    return out
+    # Calculate the sum of positive values in each row
+    df['X'] = df.drop(columns='X', errors='ignore').apply(lambda row: row[row > 0].sum(), axis=1)
+
+    # Identify rows where the number of positive values is below the threshold
+    mask_below_threshold = df.drop(columns='X').gt(0).sum(axis=1) <= degeneracy
+
+    # Reset 'X' to zero for rows below the threshold
+    df.loc[mask_below_threshold, 'X'] = 0
+
+    # Identify rows where the number of positive values exceeds the threshold
+    mask_above_threshold = ~mask_below_threshold
+
+    # Set all other values to zero for rows exceeding the threshold
+    df.loc[mask_above_threshold, df.columns != 'X'] = 0
+    # create Logo object
+    fig = logomaker.Logo(df,
+                        figsize=(0.5*splice,5),
+                        stack_order='small_on_top',
+                        color_scheme=color_scheme,
```

Comment:
As i only get the black letters, i think you need to reconsider how "color_scheme" is called.

Reply:
as discussed, they are black when we don't provide a color scheme (-cs) or if we don't provide a custom config file with the colors (-c)